### PR TITLE
[IntegratedTest] Add batch normalization test 

### PR DESCRIPTION
**Pull request by zhoonit**
~- **(Waiting for #669)** [Test] Refactor recorder.py~
~- **(Waiting for #677)** [Fix/bn] Fix batchnormalize layer~
- [Test/Util] Update ini / debug info
```
**Changes proposed in this PR:**
- Add some helperfunction to iniTestWrapper
- Add color and adam debug info to recorder.py
```
- [IntegratedTest] Add batch normalization test
```
Add test that using batchnormalization with a minor structural change on
the tester

**Changes proposed in this PR for tester:**
- Implement reorder logic for bn layer in `recorder.py`
- Add color to some debug prints
- Let layer update inside actual `backwarding`
- save weights including non_trainable but save gradients only for the
trainables

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>
```

### CODE REVIEW

File: test/input_gen/recorder.py [hunk 1]
```diff
@@ -201,17 +217,20 @@ def generate_recordable_model(loss_fn_str, model=None, inputs=None, outputs=None
 
     # omit last activation layer if cross softmax or corss_sigmoid
     if loss_fn_str == "cross_softmax" or loss_fn_str == "cross_sigmoid":
-      if isinstance(model.layers[-1], K.layers.Activation):
-        outputs = outputs[:-1]
+        if isinstance(model.layers[-1], K.layers.Activation):
```
Comment (jijoongmoon) [inline test/input_gen/recorder.py:220]: Could you check this indent?

File: test/input_gen/genModelTests.py [hunk 2]
```diff
@@ -54,8 +59,35 @@
...
+    )
+
+    fc_bn_sigmoid_tc(
+        file_name="fc_bn_sigmoid_cross.info",
+        loss_fn_str="cross_softmax",
+        # debug=["summary", "iteration", "loss"],
+    )
+
+    fc_bn_sigmoid_tc(
+        file_name="fc_bn_sigmoid_mse.info", loss_fn_str="mse",
```
Comment (kparichay) [inline test/input_gen/genModelTests.py:92]: can you please elaborate on what does a `.info` file is expected to contain?

Comment (zhoonit) [reply test/input_gen/genModelTests.py:107]: Done on the doc for the file :)

**Review by kparichay — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM
